src/Optional/Plots.jl

```diff
@@ -0,0 +1,33 @@
+using .Plots
+
+preprocess(name, val::Plots.Plot, data) = push!(data, name=>val)
+summary_impl(name, value::Plots.Plot) = image_summary(name, value)
+
+function image_summary(name::AbstractString, img::Plots.Plot)
```

Comment:
Do you think it would be better do all this in `preprocess` so as to generatre a `AbstractArray{<:Colorant}` which would just be naturally handled by the thing for plotting images?

Reply:
I did this for PyPlot, which works well, because I didn't `summary_impl` before checking this PR.

Reply:
I'd prefer this too, but how do I convert a PNG to `AbstractArray{<:Colorant}`?

Related: In [`image_summary (, img:: AbstractArray{<:Colorant})`](https://github.com/PhilipVinc/TensorBoardLogger.jl/blob/a8f4a2728a4ddbcbc7dc1d0ecbb7e5ddfa1320e5/src/Loggers/LogImage.jl#L118) we convert the array to png anyway. 
Do we really want to do `Plots->PNG->Colorant->PNG` ? 

Maybe we should invert the stack: add a PNG data type, implement `image_summary` for that, and put everything else in `preprocess`.

Reply:
That seems legit.can use the mimetype maybe.

Reply:
Mimetype to do what?

Reply:
As the type maker.
Or a FileIO.jl Format type,

Reply:
This is how I do for PyPlot.jl. Not sure is there is some similar functions in Plots.jl

```julia
using TensorBoardLogger, Logging, Images, PyCall
const mpl = PyCall.PyNULL()
const plt = PyCall.PyNULL()
copy!(mpl, PyPlot.matplotlib)
copy!(plt, mpl.pyplot)
plt_backend_agg = mpl.backends.backend_agg

function figure_to_image(fig::PyPlot.Figure; close=true)
    canvas = plt_backend_agg.FigureCanvasAgg(fig)
    canvas.draw()
    data = canvas.buffer_rgba() ./ 255
    w, h = fig.canvas.get_width_height()
    img = [RGBA(data[r,c,:]...) for r in 1:h, c in 1:w]
    if close plt.close(fig) end
    return img
end

TensorBoardLogger.preprocess(name, fig::PyPlot.Figure, data) = push!(data, name => figure_to_image(fig))
```